tls/s2n_kem.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include "tls/s2n_kem_params.h"
+
+#define KEM_EXTENSION_BYTES 1
+#define KEM_PUBLIC_KEY_BYTES 2
+
+struct s2n_kem {
+    uint8_t kem_extension_id;
+    const uint16_t publicKeySize;
+    const uint16_t privateKeySize;
+    const uint16_t sharedSecretKeySize;
+    const uint16_t ciphertextSize;
+    int (*generate_keypair)(unsigned char *public_key, unsigned char *private_key);
+    int (*encrypt)(unsigned char *ciphertext, unsigned char *shared_secret,  const unsigned char *public_key);
+    int (*decrypt)(unsigned char *shared_secret, const unsigned char *ciphertext, const unsigned char *private_key);
+};
+
+extern int s2n_kem_generate_key_pair(const struct s2n_kem *kem, struct s2n_kem_params *kem_params);
```

Comment:
Note that s2n_kem_generate_key_pair/encrypt/decrypt receive both s2n_kem and s2n_kem_params where kem_params include a pointer to s2n_kem. 
When using these functions you perform a redundant assignment:
`kem = kem_params.negotiated_kem`
and then 
`func(kem, kem_params)`

Is there a specific reason? or maybe consider leaving only the kem_params variable.

Reply:
Good point, there isn't. I think this is a left over from some refactoring that moved negotiated_kem around. I will clean this up.